.github/workflows/milestone.yml

```diff
@@ -37,7 +37,7 @@ jobs:
             const { MILESTONE_NUMBER } = process.env
 
             // Find milestone
-            const response = await github.issues.listMilestones(context.repo)
+            const response = await github.issues.listMilestones({ owner: context.repo.owner, repo: context.repo.repo, per_page: 100, })
```

Comment:
How about adding also other filters to attempt to make the results list more stable?
```suggestion
            const response = await github.issues.listMilestones({
              owner: context.repo.owner,
              repo: context.repo.repo,
              per_page: 100,
              sort: 'due_on',
              direction: 'asc',
              state: 'all'
            })
```
This would work better if milestones were created with a due date, doesn't have to be accurate and match actual releases, just for ordering. Maybe we can create them with 1 week from now.


https://octokit.github.io/rest.js/v18#issues-list-milestones

Reply:
Or if this is supposed to only find the current milestone, there should be only one open, and we can use `state: 'open'`.

Reply:
It tries to find all milestones whose title matches `MILESTONE_NUMBER` env var.

Reply:
I believe the current version uses `state: 'open'` (this is the default when nothing is specified).

Reply:
Here's the default: https://docs.github.com/en/rest/issues/milestones#list-milestones

Reply:
So is it a race?

Reply:
Maybe Github has some internal index for milestones that don't get refreshed immediately after creating a new one and it doesn't return it, but the subsequent create call properly detects it already exists. I'll see if the PRs where it failed were created after the release.

Reply:
Ok, so if there's only one open, I don't think it's a proper fix.
Maybe it is a race condition, so we should just retry?

Reply:
It happens so rarely it's hard to confirm :-) The question is how many times to retry or how long we should keep trying. Doesn't look like we can get the current milestone in any other way :-(

Reply:
I'll close this PR, as it change doesn't make sense.